src/doc/unstable-book/src/compiler-flags/remap-path-prefix.md

```diff
@@ -0,0 +1,37 @@
+# `remap-path-prefix`
+
+The tracking issue for this feature is: [#41555](https://github.com/rust-lang/rust/issues/41555)
+
+------------------------
+
+The `-Z remap-path-prefix-from`, `-Z remap-path-prefix-to` commandline option
+pair allows to replace prefixes of any file paths the compiler emits in various
+places. This is useful for bringing debuginfo paths into a well-known form and
+for achieving reproducible builds independent of the directory the compiler was
+executed in. All paths emitted by the compiler are affected, including those in
+error messages.
+
+In order to map all paths starting with `/home/foo/my-project/src` to
+`/sources/my-project`, one would invoke the compiler as follows:
+
+```text
+rustc -Zremap-path-prefix-from="/home/foo/my-project/src" -Zremap-path-prefix-to="/sources/my-project"
+```
+
```

Comment:
It's unclear from this how the remapping is actually operating. It should also answer:

- Is it delimited by path element boundaries, or is it operating on literal strings?
- Is there any normalization performed before/after doing the transformation? (Redundant `/` elimination?)
- Are the input paths absolute, relative, or whatever was provided to the compiler?
- In this example, if the input path were `/home/foo/my-project/src-orig`, would this remap to `/sources/my-project-orig`? Should there be a trailing `/` to prevent this if its undesirable?